backgroundfield/backgroundfield.h

```diff
@@ -23,114 +23,86 @@
 #ifndef BACKGROUNDFIELD_H
 #define BACKGROUNDFIELD_H
 
-#include "fieldfunction.hpp"
-#include "integratefunction.hpp"
-#include "../definitions.h"
 #include "../common.h"
+#include "../definitions.h"
+#include "fieldfunction.hpp"
 #include "fsgrid.hpp"
+#include "integratefunction.hpp"
+#include "phiprof.hpp"
+#include <span>
 
-void setBackgroundField(
-   const FieldFunction& bgFunction,
-   FsGrid< std::array<Real, fsgrids::bgbfield::N_BGB>, FS_STENCIL_WIDTH> & BgBGrid,
-   bool append=false
-);
+void setBackgroundField(const FieldFunction& bgFunction, std::span<std::array<Real, fsgrids::bgbfield::N_BGB>> bgb,
+                        std::span<fsgrids::technical> technical, FieldSolverGrid &fsgrid, bool append = false);
 
-void setBackgroundFieldToZero(
-   FsGrid< std::array<Real, fsgrids::bgbfield::N_BGB>, FS_STENCIL_WIDTH> & BgBGrid
-);
+void setBackgroundFieldToZero(std::span<std::array<Real, fsgrids::bgbfield::N_BGB>> bgb);
 
 /**
    Templated function for setting the perturbed B field to zero.
    Function is templated so it may, if necessary, be called to set the
    vector dipole correction terms inside the background field FSgrid
    object to zero (see setPerturbedField below).
 */
-template<long unsigned int numFields> void setPerturbedFieldToZero(
-   FsGrid< std::array<Real, numFields>, FS_STENCIL_WIDTH> & BGrid,
-   int offset=fsgrids::bfield::PERBX
-   ) {
-   std::array<FsGridTools::FsIndex_t,3> localSize = BGrid.getLocalSize();
-
-   #pragma omp parallel for collapse(2)
-   for (FsGridTools::FsIndex_t z = 0; z < localSize[2]; ++z) {
-      for (FsGridTools::FsIndex_t y = 0; y < localSize[1]; ++y) {
-         for (FsGridTools::FsIndex_t x = 0; x < localSize[0]; ++x) {
-            // This is still N_BFIELD (==3) instead of numFields
-            for (int i = 0; i < fsgrids::bfield::N_BFIELD; ++i) {
-               BGrid.get(x,y,z)->at(offset+i) = 0;
-            }
-         }
+template <long unsigned int numFields>
+void setPerturbedFieldToZero(std::span<std::array<Real, numFields>> b, int offset = fsgrids::bfield::PERBX) {
+#pragma omp parallel for
```

Comment:
Ah, same here? I had grepped for nested loops, but I guess this should then also be ported?